File: frontend/src/@types/convocacaoExamesGeral.ts
export type ConvocacaoExamesGeral = {
	cargo: string | null;
	nome: string | null;
	matricula: string | null;
	exame: string | null;
	ultimo_pedido: string | null;
	data_resultado: string | null;
	periodicidade: number | null;
	refazer_em: string | null;
	dt_admissao: string | null;
	situacao: string | null;
	cidade_unidade: string | null;
	estado_unidade: string | null;
	dt_nascimento: string | null;
	data_referencia: string | null;
};

export type ConvocacaoExamesGeralQuery = {
	page?: number;
	size?: number;
	sort_by?: "matricula" | "dt_admissao";
	sort_order?: "asc" | "desc";
	cargo?: string;
	nome?: string;
	matricula?: string;
	dt_admissao__gte?: string;
	dt_admissao__lte?: string;
	situacao?: string;
	dt_nascimento_gte?: string;
	dt_nascimento_lte?: string;
};

File: backend/src/schemas/responses/convocacao_exames_geral.py
from datetime import date
from typing import Optional

from pydantic import BaseModel


class ConvocacaoDeExamesGeralResponse(BaseModel):
    cargo: Optional[str]
    nome: Optional[str]
    matricula: Optional[str]
    exame: Optional[str]
    ultimo_pedido: Optional[str]
    data_resultado: Optional[date]
    periodicidade: Optional[int]
    refazer_em: Optional[str]
    dt_admissao: Optional[date]
    situacao: Optional[str]
    cidade_unidade: Optional[str]
    estado_unidade: Optional[str]
    dt_nascimento: Optional[date]
    data_referencia: Optional[date]
